ansible/configs/kni-osp/files/cloud_providers/osp_cloud_template_master.j2

```diff
@@ -0,0 +1,262 @@
+heat_template_version: 2018-03-02
+
+description: Top level HOT for creating new project, network resources, and instances. This template relies on ResourceGroups and a nested template that is called to provision instances, ports, & floating IPs.
+
+resources:
+
+  {{ guid }}-infra_key:
+    type: OS::Nova::KeyPair
+    properties:
+      name: {{ guid }}-infra_key
+      save_private_key: True
+
+{% if osp_project_create | bool %}
```

Comment:
This needs to be remove as well as the project_role_user. These are taken care of in a role prior to generating these HOTs.

Reply:
are you sure ? It's in the default template: https://github.com/redhat-cop/agnosticd/blob/development/ansible/roles-infra/infra-osp-template-generate/templates/cloud_template_master.j2#L12-L16

The users are created outside of the stack now, and the stack is executed using the user, not the admin. But the key is still created here in the stack  IIUC

Reply:
Ok i understand now, Nate is referring to the portion of code after:

```
{% if osp_project_create | bool %}
```

like this: https://github.com/redhat-cop/agnosticd/commit/d83cfbea76ae2e8d9cce0ee36e84cbdc48402e0b#diff-9b5c2aebe156053fa12a54d05746a91e

Reply:
removed that one now :)